contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,370 @@
+        if (callType == ERC7579Utils.CALLTYPE_BATCH) return executionCalldata.execBatch(execType);
+        if (callType == ERC7579Utils.CALLTYPE_DELEGATECALL) return executionCalldata.execDelegateCall(execType);
+        revert ERC7579Utils.ERC7579UnsupportedCallType(callType);
+    }
+
+    /**
+     * @dev Lowest-level signature validation function. See {ERC7739Signer-_rawSignatureValidation}.
+     *
+     * This function delegates the signature validation to a validation module if the first 20 bytes of the
+     * signature correspond to an installed validator module.
+     *
+     * See {_extractSignatureValidator} for the module extraction logic.
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
+        if (signature.length < 20) return false;
+        address module = _extractSignatureValidator(signature);
+        return
+            isModuleInstalled(MODULE_TYPE_VALIDATOR, module, msg.data) &&
+            IERC7579Validator(module).isValidSignatureWithSender(address(this), hash, signature[20:]) ==
+            IERC1271.isValidSignature.selector;
+    }
+
+    /// @dev Checks if the module is installed. Reverts if the module is not installed.
+    function _checkModule(uint256 moduleTypeId, address module) internal view virtual {
+        require(
+            isModuleInstalled(moduleTypeId, module, msg.data),
+            ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module)
+        );
+    }
+
+    /**
+     * @dev Installs a module of the given type with the given initialization data.
+     *
+     * For the fallback module type, the `initData` is expected to be a tuple of a 4-byte selector and the
+     * rest of the data to be sent to the handler when calling {IERC7579Module-onInstall}.
+     *
+     * Requirements:
+     *
+     * * Module type must be supported. See {supportsModule}. Reverts with {ERC7579UnsupportedModuleType}.
+     * * Module must be of the given type. Reverts with {ERC7579MismatchedModuleTypeId}.
+     * * Module must not be already installed. Reverts with {ERC7579AlreadyInstalledModule}.
+     *
+     * Emits a {ModuleInstalled} event.
+     */
+    function _installModule(uint256 moduleTypeId, address module, bytes memory initData) internal virtual {
+        require(supportsModule(moduleTypeId), ERC7579Utils.ERC7579UnsupportedModuleType(moduleTypeId));
+        require(
+            IERC7579Module(module).isModuleType(moduleTypeId),
+            ERC7579Utils.ERC7579MismatchedModuleTypeId(moduleTypeId, module)
+        );
+
+        bool exists;
+        if (moduleTypeId == MODULE_TYPE_VALIDATOR) exists = !_validators.add(module);
+        if (moduleTypeId == MODULE_TYPE_EXECUTOR) exists = !_executors.add(module);
+        require(!exists, ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module));
+
+        if (moduleTypeId == MODULE_TYPE_FALLBACK) {
```

Comment:
IMO this would be clearer (also it avoid the double negation)

```suggestion
        if (moduleTypeId == MODULE_TYPE_VALIDATOR) {
            require(_validators.add(module), ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module);
        } else if (moduleTypeId == MODULE_TYPE_EXECUTOR) {
            require(_executors.add(module) , ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module);
        } else if (moduleTypeId == MODULE_TYPE_FALLBACK) {
```